--- a/fs/proc/cmdline.c
+++ b/fs/proc/cmdline.c
@@ -4,6 +4,9 @@
 #include <linux/proc_fs.h>
 #include <linux/seq_file.h>
 #include <asm/setup.h>
+#ifdef CONFIG_KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG
+extern int susfs_spoof_cmdline_or_bootconfig(struct seq_file *m);
+#endif
 
 static char new_command_line[COMMAND_LINE_SIZE];
 
@@ -11,6 +14,12 @@ static int cmdline_proc_show(struct seq_file *m, void *v)
 {
 	seq_printf(m, "%s\n", new_command_line);
 	return 0;
+#ifdef CONFIG_KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG
+	if (!susfs_spoof_cmdline_or_bootconfig(m)) {
+		seq_putc(m, '\n');
+		return 0;
+	}
+#endif
 }
 
 static int cmdline_proc_open(struct inode *inode, struct file *file)
